plugins/amongus/amongus.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "Game.h"
+
+#include "mumble_plugin.h"
+#include "mumble_plugin_utils.h"
+
+#include <sstream>
+
+std::unique_ptr< Game > game;
+
+static inline bool inGame(const AmongUsClient_Fields &fields) {
+	if (fields.gameMode != GameMode::FreePlay && fields.gameState != GameState::NotJoined) {
+		return true;
+	}
+
+	return false;
+}
+
+static int fetch(float *avatarPos, float *avatarFront, float *avatarTop, float *cameraPos, float *cameraFront,
+				 float *cameraTop, std::string &context, std::wstring &identity) {
+	for (uint8_t i = 0; i < 3; ++i) {
+		avatarPos[i] = avatarFront[i] = avatarTop[i] = cameraPos[i] = cameraFront[i] = cameraTop[i] = 0.f;
+	}
+
+	const auto fields = game->clientFields();
+
+	if (!inGame(fields)) {
+		return false;
+	}
+
+	if (fields.gameState == GameState::Ended) {
+		return true;
+	}
+
+	const auto playerControlFields = game->playerControlFields();
+
+	// Mumble | Game
+	// X      | X
+	// Y      | -
+	// Z      | Y
+	const auto position = game->playerPosition(playerControlFields.netTransform);
+	avatarPos[0] = cameraPos[0] = position.x;
+	avatarPos[2] = cameraPos[2] = position.y;
+
+	// Context
+	std::ostringstream stream;
+	stream << " {\"Game ID\": " << fields.gameId << "}";
+	context = stream.str();
+
+	stream.str(std::string());
+	stream.clear();
+
+	// Identity
+	const auto playerFields = game->playerInfoFields(playerControlFields.cachedData);
+
+	stream << "Name: " << game->string(playerFields.playerName) << std::endl;
+	stream << "Client ID: " << std::to_string(fields.clientId) << std::endl;
+	stream << "Player ID: " << std::to_string(playerFields.playerId) << std::endl;
+	stream << "Color ID: " << std::to_string(playerFields.colorId) << std::endl;
+	stream << "Skin ID: " << std::to_string(playerFields.skinId) << std::endl;
+	stream << "Hat ID: " << std::to_string(playerFields.hatId) << std::endl;
+	stream << "Pet ID: " << std::to_string(playerFields.petId) << std::endl;
+	stream << "Dead: " << (playerFields.isDead ? "true" : "false") << std::endl;
+	stream << "Impostor: " << (playerFields.isImpostor ? "true" : "false") << std::endl;
```

Comment:
I'm considering removing this, as it would allow anyone who can read the identity to cheat.

Reply:
Well I think the positional data alone is enough info for being able to cheat. Not sure if we have to be this cautious here :thinking:

Reply:
Technically the positional data could indeed indicate whether a player is an impostor, for example due to the coordinates "jumping" when moving through a duct (only impostors can do that).

However, I'm pretty sure it can be mistaken for lag.